framework/framework.go

```diff
@@ -142,3 +142,60 @@ func (f *Framework) ProcessDelete(obj interface{}, resources []Resource) error {
 
 	return nil
 }
+
+// ProcessUpdate is a drop-in for an informer's UpdateFunc. It receives the
+// custom object observed during TPR watches and anything that implements
+// Resource. ProcessUpdate takes care about all necessary reconciliation logic
+// for update events. For complex resources this means state has to be created,
+// deleted and updated eventually, in this order.
+//
+//     func updateFunc(oldObj, newObj interface{}) {
+//         err := f.ProcessUpdate(oldObj, newObj, resources)
+//         if err != nil {
+//             // error handling here
+//         }
+//     }
+//
+//     newResourceEventHandler := &cache.ResourceEventHandlerFuncs{
+//         UpdateFunc:    updateFunc,
+//     }
+//
+func (f *Framework) ProcessUpdate(oldObj, newObj interface{}, resources []Resource) error {
```

Comment:
I don't know what you mean by `separation of concerns`, because when an operator receives `OnAdd` event it *never* knows if this is really newly created TPO or just a replayed event (e.g. listing after `ResyncPeriod`, or restarted operator pod).